community/contributing/pr_review_guidelines.rst

```diff
@@ -0,0 +1,335 @@
+
+5. Iterate with the contributor and improve the PR
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Inevitably, a lot of pull requests would be initially undercooked.
+At this point, you and the contributor should enter a feedback loop
+where they iterate on your notes, recommendations and requests, and
+you look that they have indeed fixed the issues you've highlighted.
+
+Try not to exhaust the contributor with style nitpicks, especially
+if it's still up in the air whether their PR would be accepted at
+all. Preferably, suggestions should come in order of importance:
+first address their overall code design and approach to solving the
+problem, then make sure their code is complying with the engine's
+best practices, and last do the "pretty pass".
+
+Some areas of the engine are more important than others. There is
+a lot of ugly code in the editor and UI components, and while this
+is not ideal, it's not the end of the world either. It's fine to
+suggest improvements there as well, but don't insist too much if
+the end result is achieved and the contribution leaves the code
+reasonably maintainable.
+
+And ultimately, don't feel pressured to deal with the PR all
+alone. Feel free to ask for a helping hand on the `Godot
+Contributors Chat <https://chat.godotengine.org>`_, in the appropriate
+channel or in general. Other teams may already be tagged for review,
+so you can also wait or ask for their assistance.
+
+6. Approve the pull request
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you find that everything is in order, that the PR is addressing
+a problem and does so in the acceptable way (it doesn't have to be
+perfect, but it helps), then put your positive review on it.
+
+If you are a merger
+-------------------
+
+A merger is just a reviewer in fancy pants. All recommendations
+above still apply if you aim to merge a pull request, but you
+can also completely rely on your fellow maintainers and their
+judgement. Naturally, there is little point in doing a full review
+yourself if you see your team members approving the changes.
+
+But there are still more things that need to be checked before a
+PR can be merged.
+
+1. Get feedback from multiple people/teams
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Try not to merge things based on one review alone, especially
+your own. Get a second opinion from your teammates, and make
+sure all the teams have been reasonably represented by the
+reviewers. For example, if a pull request adds to the documentation,
+it's often useful to let the area maintainers check it for
+factual correctness and let documentation maintainers check it
+for formatting, style, and grammar.
+
+Make sure that the reviews and approvals were left by people
+competent in that specific engine area. It is possible that
+even a long-standing member of the Godot organization left
+a review without having the relevant expertise. As a merger
+you can overrule their approving decision or their requests
+for changes.
```

Comment:
AFAIK when possible the people merging should be area maintainers and members of a team maintaining that area. I don't know how strict this is but I know I've seen this mentioned in the past so we should mention it somewhere. The current text here talks about competent-in-the-area people reviewing and then someone else merging.

Reply:
Hmm, I don't think that I've made any statement in the article that implies you should review and merge just about anything. But I guess I also didn't make a clear statement that you should focus on issues you yourself are competent at (namely, PRs that await a review from the team you are in).

Throughout the document it is implied that you didn't stumble upon the issue randomly, but were assigned to it via the team you are in.